kubectl-volsync/cmd/migration_create.go

```diff
@@ -36,70 +79,321 @@ var migrationCreateCmd = &cobra.Command{
+			return nil, err
+		}
+		mc.PVC = PVC
+	} else {
+		klog.Infof("Destination PVC: \"%s\" is found in Namespace: \"%s\" and is used to create replication destination",
+			mc.PVC.Name, mc.PVC.Namespace)
+	}
+
+	return mc.PVC, nil
+}
+
+func (mc *migrationCreate) createDestinationPVC(ctx context.Context) (*corev1.PersistentVolumeClaim, error) {
+	destPVC := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mc.DestinationPVC,
+			Namespace: mc.Namespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      mc.AccessModes,
+			StorageClassName: mc.StorageClassName,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: *mc.Capacity,
+				},
+			},
+		},
+	}
+
+	if err := mc.clientObject.Create(ctx, destPVC); err != nil {
+		return nil, err
+	}
 
-	// TODO: Set up objects on the cluster
-	_, _ = newClient("")
+	klog.Infof("Created Destination PVC: \"%s\" in NameSpace: \"%s\" and Cluster: \"%s\" ",
+		destPVC.Name, destPVC.Namespace, destPVC.ClusterName)
+
+	return destPVC, nil
+}
+
+func (mc *migrationCreate) getDestinationPVC(ctx context.Context) (*corev1.PersistentVolumeClaim, error) {
+	destPVC := &corev1.PersistentVolumeClaim{}
+	pvcInfo := types.NamespacedName{
+		Namespace: mc.Namespace,
+		Name:      mc.DestinationPVC,
+	}
+	err := mc.clientObject.Get(ctx, pvcInfo, destPVC)
+	if err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			klog.Infof("pvc: \"%s\" not found, creating the same", mc.DestinationPVC)
+			return nil, nil
+		}
+		return nil, err
+	}
+	return destPVC, nil
+}
+
+func (mc *migrationCreate) ensureReplicationDestination(ctx context.Context) (
+	*volsyncv1alpha1.ReplicationDestination, error) {
+	mrd := mc.mr.data.Destination
+	rd := &volsyncv1alpha1.ReplicationDestination{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mrd.RDName,
+			Namespace: mrd.Namespace,
+		},
+		Spec: volsyncv1alpha1.ReplicationDestinationSpec{
+			Rsync: &volsyncv1alpha1.ReplicationDestinationRsyncSpec{
+				ReplicationDestinationVolumeOptions: volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+					DestinationPVC: mrd.Destination.DestinationPVC,
+				},
+				ServiceType: mrd.Destination.ServiceType,
+			},
+		},
+	}
+	if err := mc.clientObject.Create(ctx, rd); err != nil {
+		return nil, err
+	}
+	klog.Infof("Created ReplicationDestination: \"%s\" in Namespace: \"%s\" and Cluster: \"%s\"",
+		rd.Name, rd.Namespace, rd.ClusterName)
+
+	return rd, nil
+}
+
+func (mc *migrationCreate) waitForRDStatus(ctx context.Context) error {
+	mrd := mc.mr.data.Destination
+	// wait for migrationdestination to become ready
+	nsName := types.NamespacedName{
+		Namespace: mrd.Namespace,
+		Name:      mrd.RDName,
+	}
+	rd := &volsyncv1alpha1.ReplicationDestination{}
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		err := mc.clientObject.Get(ctx, nsName, rd)
+		if err != nil {
+			return false, err
+		}
+		if rd.Status == nil {
```

Comment:
Rsync field is a pointer, so need a `nil` check

```suggestion
		if rd.Status == nil || rd.Status.Rsync == nil {
```